nat/mapping/port_mapping.go

```diff
@@ -64,7 +67,14 @@ func mapPort(m portmap.Interface, c chan struct{}, protocol string, extPort, int
 		}
 	}()
 	for {
-		addMapping(m, protocol, extPort, intPort, name)
+		err := addMapping(m, protocol, extPort, intPort, name)
+		if err != nil {
+			log.Infof("%s, Mapping for port %d failed: %s", logPrefix, extPort, err)
+			metricsSender.SendNATMappingFailEvent(err)
+		} else {
+			log.Info("%s, Mapped network port: %d", logPrefix, extPort)
+			metricsSender.SendNATMappingSuccessEvent()
```

Comment:
Theres one bad issue with this implementation. Since mapPort is executed in routine every `mapUpdateInterval` this means we will be getting as much statistics updates. This might be overhelming for out metrics collector. For NAT statistics analysis it should be sufficient for us to see just a single report after service gets started. 

My suggestion as to decouple from metrics sender here and to solve this problem is to use an EventBus.

I am working currently introducing event bus to port_mapper for communication with other NAT submodules, but these could be reused them. I would suggent to stall this PR and to make it work for both NAT methods reusing EventBus.

Reply:
I had similar ideas in mind here also https://github.com/mysteriumnetwork/node/pull/762#discussion_r262380471